src/screens/Ticketing/Ticket/TicketInfo.tsx

```diff
@@ -0,0 +1,98 @@
+import {FareContract, FareContractTraveller} from '../../../api/fareContracts';
+import ThemeText from '../../../components/text';
+import {TicketTexts, useTranslation} from '../../../translations';
+import React from 'react';
+import {View} from 'react-native';
+import {useRemoteConfig} from '../../../RemoteConfigContext';
+import {UserProfile} from '../../../reference-data/types';
+
+const TicketInfo = ({fareContract: fc}: {fareContract: FareContract}) => {
+  const {t} = useTranslation();
+  const {
+    tariff_zones: tariffZones,
+    preassigned_fare_products: preassignedFareProducts,
+    user_profiles: userProfiles,
+  } = useRemoteConfig();
+
+  const preassignedFareProduct = findById(
+    preassignedFareProducts,
+    fc.travellers[0].fare_product_ref,
+  );
+
+  const fromTariffZone = findById(
+    tariffZones,
+    fc.travellers[0].tariff_zone_refs[0],
+  );
+  const toTariffZone = findById(
+    tariffZones,
+    fc.travellers[0].tariff_zone_refs.slice(-1)[0],
+  );
```

Comment:
Suggested refactor for readability:

```suggestion
  const {fare_product_ref: productRef, tariff_zone_refs} = fc.travellers[0];
  const [firstZone] = tariff_zone_refs;
  const [lastZone] = tariff_zone_refs.slice(-1);

  const preassignedFareProduct = findById(preassignedFareProducts, productRef);
  const fromTariffZone = findById(tariffZones, firstZone);
  const toTariffZone = findById(tariffZones, lastZone);
```